src/main/java/io/xpring/xrpl/model/XrpCheckCreate.java

```diff
@@ -0,0 +1,116 @@
+package io.xpring.xrpl.model;
+
+import io.xpring.common.XrplNetwork;
+import io.xpring.xrpl.ClassicAddress;
+import io.xpring.xrpl.ImmutableClassicAddress;
+import io.xpring.xrpl.Utils;
+import org.immutables.value.Value;
+import org.xrpl.rpc.v1.CheckCreate;
+
+import java.util.Optional;
+
+/**
+ * Represents a CheckCreate transaction on the XRP Ledger.
+ * <p>
+ * A CheckCreate transaction creates a Check object in the ledger, which is a deferred payment that can be cashed
+ * by its intended destination.  The sender of this transaction is the sender of the Check.
+ * </p>
+ *
+ * @see "https://xrpl.org/checkcreate.html"
+ */
+@Value.Immutable
+public interface XrpCheckCreate {
+  static ImmutableXrpCheckCreate.Builder builder() {
+    return ImmutableXrpCheckCreate.builder();
+  }
+
+  /**
+   * The unique address and (optional) destination tag of the account that can cash the Check,
+   * encoded as an X-address.
+   *
+   * @return A {@link String} representing the unique address and (optional) destination tag of the account that
+   *         can cash the Check, encoded as an X-address.
+   *
+   * @see "https://xrpaddress.info/)"
+   */
+  String destinationXAddress();
+
+  /**
+   * (Optional) Time after which the Check is no longer valid, in seconds since the Ripple Epoch.
+   *
+   * @return An {@link Integer} representing the time after which the Check is no longer valid, in seconds
+   *         since the Ripple Epoch.
+   */
+  Optional<Integer> expiration();
+
+  /**
+   * (Optional) Arbitrary 256-bit hash representing a specific reason or identifier for this Check.
+   *
+   * @return A {@link String} containing an arbitrary 256-bit hash representing a specific reason or identifier for
+   *         this Check.
+   */
+  Optional<String> invoiceID();
```

Comment:
the case of the protobuf getter is `invoiceId` but it seems to be `invoiceID` in other places?

Reply:
the casing in the JS PR is `invoiceId` and it's idiomatic in Java to present abbreviations in camel case as well... so I think we should probably call this field `invoiceId`

Reply:
got it, that makes sense. i switched to `invoiceId`